lib/precondition.js

```diff
@@ -1,14 +1,15 @@
 var MAX_ALLOC = Math.pow(2, 30) - 1 // default in iojs
 
-function checkBuffer (buf, name) {
-  if (typeof buf !== 'string' && !Buffer.isBuffer(buf)) {
+function checkType (variable, name) {
+  // checks whether the type of 'variable' is one of string, Buffer or Uint8Array
+  if (typeof buf !== 'string' && !Buffer.isBuffer(buf) && !buf.constructor.name === 'Uint8Array') {
```

Comment:
this will error if it's undefined [this](https://github.com/nodejs/node/blob/b2f0cfa6b0e139a2f990d4e1e7104abf015fe8e7/lib/stream.js#L59) is how node does it